### PR TITLE
[doc] Update i2c documentation.

### PR DESCRIPTION
**Pull request by tjaychen**
Matches recent target stretch usability updates.

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

File: hw/ip/i2c/doc/_index.md [hunk 1]
```diff
@@ -255,31 +252,26 @@ Other features may also be required for complete SMBus functionality.)
...
 
-In all cases described above, a target begins to stretch the clock when SCL is held low by a host.
+#### Stretching during read
+    - When a target device receives a start and read command, it may stretch the clock for either of the following two reasons.
+      - If there is no data available to be sent back (TX FIFO empty case), the target stretches the clock until data is made available by software.
+      - If there is more than 1 entry in the ACQ FIFO.
+        - Having more than 1 entry in the ACQ FIFO suggests there is potentially an unhandled condition (STOP / RESTART) or an unhandled command (START) that requires software intervention before the read can proceed.
+    - The `tx_stretch` interrupt is generated to alert software to such a situation.
+
+In all cases described above, a target begins to stretch the clock after the ACK bit.
```
Comment (a-will) [inline hw/ip/i2c/doc/_index.md:273]: This is going to look like it's under the "Stretching during read" label, isn't it? If so, maybe this text should be adjusted to go above the list, or the labels should become part of the list instead?

**Review by a-will — APPROVED** [review]
LGTM, with one question about rendering.

File: hw/ip/i2c/doc/_index.md (hunk 1 above)
Comment (tjaychen) [reply hw/ip/i2c/doc/_index.md:273]: o yeah good point.

File: hw/ip/i2c/doc/_index.md [hunk 2]
```diff
@@ -37,12 +35,11 @@ See that document for integration overview within the broader top level system.
 - SW may reset I2C block using the Reset Manager.
 
 <sup>1</sup> lowRISC is avoiding the fraught terms master/slave and defaulting to host/target where applicable.
-
-<sup>2</sup> The target is only compatible with hosts that support clock stretching.
+<sup>2</sup> The target is only compatible with hosts that support clock stretching. For hosts that do not support clock stretching, it is expected that there must be an additional protocol to guarantee there is always sufficient space and data.  These protocols are not in scope of this document.
```
Comment (marnovandermaas) [inline hw/ip/i2c/doc/_index.md:38]: Nit: each sentence should have its own line.

File: hw/ip/i2c/doc/_index.md [hunk 3]
```diff
@@ -187,7 +184,7 @@ The following diagram shows consecutive entries inserted into ACQ FIFO during a
 ![](I2C_acq_fifo_write.svg)
 
 If the transaction is a read operation (R/W bit = 1), the target pulls bytes out of TX FIFO and transmits them to the bus until the host signals the end of the transfer by sending a NACK signal.
-If TX FIFO holds no data, the target will hold SCL low to stretch the clock and give software time to write data bytes into TX FIFO.
+If TX FIFO holds no data, or if the ACQ FIFO contains more than 1 entry, and thus a potentially unhandled command from the host, the target will hold SCL low to stretch the clock and give software time to write data bytes into TX FIFO or handle the available command.
```
Comment (marnovandermaas) [inline hw/ip/i2c/doc/_index.md:187]:
```suggestion
If TX FIFO holds no data or if the ACQ FIFO contains more than 1 entry the target will hold SCL low to stretch the clock and give software on the host time to write data bytes into TX FIFO or handle the available command.
```
Nit: maybe removing the nested clause makes the sentence less confusing?

File: hw/ip/i2c/doc/_index.md [hunk 4]
```diff
@@ -23,9 +23,7 @@ See that document for integration overview within the broader top level system.
     - 7-bit target address
 - Support for the following optional capabilities:
     - Clock stretching in the host mode
-    - Automatic clock stretching in the target mode when TX FIFO is empty during a read<sup>2</sup>
-    - Automatic clock stretching in the target mode when ACQ FIFO is full during a write
-    - Clock stretching in the target mode after a complete address acquire, byte read, or byte write transactions
+    - Automatic clock stretching in the target mode<sup>2</sup>
```
Comment (marnovandermaas) [inline hw/ip/i2c/doc/_index.md:26]: Nit: should this bullet point end in a period?

**Review by marnovandermaas — APPROVED** [review]
LGTM, just some minor nits.

File: hw/ip/i2c/doc/_index.md (hunk 4 above)
Comment (tjaychen) [reply hw/ip/i2c/doc/_index.md:26]: ah for this one i just followed the previous bullets, didn't want to make this one stand out.

**Comment by tjaychen** [comment]
I'm merging for now since this is a documentation only fix and has no impact on verilator testing.

**Pull request merged**